frontend/src/features/documents/list/DocumentListContainer.tsx

```diff
@@ -16,24 +16,38 @@ const DocumentListContainer: React.FunctionComponent<IDocumentListContainerProps
   const {
     retrieveDocumentRelationship,
     retrieveDocumentRelationshipLoading,
+    deleteDocumentRelationship,
   } = useDocumentRelationshipProvider();
+
+  const retrieveDocuments = useCallback(async () => {
+    const documents = await retrieveDocumentRelationship(props.relationshipType, props.parentId);
+    if (documents !== undefined) {
+      setDocumentResults(documents);
+    }
+  }, [retrieveDocumentRelationship, props.relationshipType, props.parentId]);
+
   useEffect(() => {
-    const fetch = async () => {
-      const documents = await retrieveDocumentRelationship(props.relationshipType, props.parentId);
-      if (documents !== undefined) {
-        setDocumentResults(documents);
-      }
-    };
+    retrieveDocuments();
+  }, [retrieveDocuments]);
 
-    fetch();
-  }, [props.parentId, props.relationshipType, retrieveDocumentRelationship]);
+  const deleteHandle = (documentRelationship: Api_DocumentRelationship): Promise<boolean> => {
+    return new Promise<boolean>((resolve, reject) => {
+      deleteDocumentRelationship(props.relationshipType, documentRelationship).then(result => {
+        if (result) {
+          retrieveDocuments();
```

Comment:
do we need to check if the user is still on the page (hasn't navigated away) here? (ismounted)